lib/mix/lib/mix/pubsub.ex

```diff
@@ -0,0 +1,87 @@
+defmodule Mix.PubSub do
```

Comment:
Should we have it under `Mix.Sync`? I actually like putting it all under the `Mix.Sync` namespace instead of root.